packages/Kokkos/src/DTK_PolyhedronList.hpp

```diff
@@ -0,0 +1,156 @@
+    //! number of nodes composing each face) or the sum of all elements in the
+    //! following view, nodes_per_face, which indicates how many nodes are
+    //! assigned to each face and how to index into this view. The input
+    //! should be arranged as follows. Consider the \f$n^th\f$ node of face
+    //! \f$i\f$ to be \f$f^i_n\f$ which is equal to the local index of the
+    //! corresponding node in the coordinates view. Two faces, the first with
+    //! 4 nodes and the second with 3 would then be defined via this view as:
+    //! \f$(f^1_1, f^1_2, f^1_3, f^1_4, f^2_1, f^2_2, f^2_3 )\f$ with the
+    //! nodes_per_face view reading \f$(4, 3)\f$
+    Kokkos::View<LocalOrdinal *, ViewProperties...> faces;
+
+    //! Number of nodes composing each face in the face input view. The sum of
+    //! all local elements in this view should equal the total size of the
+    //! faces view. This view rank-1 and should be of length of the number of
+    //! faces in the list.
+    Kokkos::View<int *, ViewProperties...> nodes_per_face;
+
+    //! Connectivity list of polyhedrons. This view is defined as rank-1 but
+    //! represents unstructured rank-2 data. It should be sized as (total sum
+    //! of the number of faces composing each polyhedron) or the sum of all
+    //! elements in the view faces_per_cells, which indicates how many faces
+    //! are assigned to each cell and how to index into this view. The input
+    //! should be arranged as follows. Consider the \f$n^th\f$ face of cell
+    //! \f$i\f$ to be \f$c^i_n\f$ which is equal to the local index of the
+    //! corresponding face in the faces view. Two cells, the first with 5
+    //! faces and the second with 4 would then be defined via this view as:
+    //! \f$(c^1_1, c^1_2, c^1_3, c^1_4, c^1_5, c^2_1, c^2_2, c^2_3, c^2_4 )\f$
+    //! with the faces_per_cell view reading \f$(5, 4)\f$.
+    Kokkos::View<LocalOrdinal *, ViewProperties...> cells;
+
+    //! Number of cells composing each cell in the cell input view. The sum of
+    //! all local elements in this view should equal the total size of the
+    //! cells view. This view is rank-1 and of length of the number of cells in
+    //! the list.
+    Kokkos::View<int *, ViewProperties...> faces_per_cell;
+
+    //! Orientation of each face composing a cell indicating an outward or
+    //! inward facing normal based on node ordering of the face and use of the
+    //! right-hand rule. This view is defined as rank-1 but represents
+    //! unstructured rank-2 data. This view is the same size as the cells view
+    //! and is indexed in an identical matter. If the face for the given cell
+    //! has a node ordering that returns a face normal that points into the
+    //! cell via the right hand rule then a 0 should be input. If the node
+    //! ordering of the face produces a normal that points out from the cell a
+    //! 1 should be input.
+    Kokkos::View<int *, ViewProperties...> face_orientation;
```

Comment:
yeah me too. ultimately i think encoding the +/- within the cells array is the best way to do it and would eliminate one of the arrays but then we couldn't use our `LocalOrdinal` type for the cells array because it is unsigned